lib/facade/FileClient.ts

```diff
@@ -14,7 +14,8 @@ interface FileClientConfig {
 }
 
 function isNodeStream(source: InputSource): source is NodeJS.ReadableStream {
-  return Object.prototype.hasOwnProperty.call(source, 'pipe');
```

Comment:
amazingly hasOwnProperty returns false for stream altough `!!source.pipe` returns pipe. HasOwnPropery is for innumerable props (the one that will appear in loop, native methods are not innumerable).